code/datums/components/gun_crank.dm

```diff
@@ -0,0 +1,59 @@
+// Cranking feature on the laser musket and smoothbore disabler, could possibly be used on more than guns
+/datum/component/gun_crank
+	/// Our cell to charge
+	var/obj/item/stock_parts/cell/charging_cell
+	/// How much charge we give our cell on each crank
+	var/charge_amount
+	/// How long is the cooldown time between each charge
+	var/cooldown_time
+	/// The sound used when charging, renember to adjust the cooldown time to keep it sensible
+	var/charge_sound
+	/// How long is the cooldown between charging sounds
+	var/charge_sound_cooldown_time
+	/// Are we currently charging
+	var/is_charging = FALSE
+	COOLDOWN_DECLARE(charge_sound_cooldown)
+
+/datum/component/gun_crank/Initialize(charging_cell, charge_amount = 500, cooldown_time = 2 SECONDS, charge_sound = 'sound/weapons/laser_crank.ogg', charge_sound_cooldown_time = 1.8 SECONDS)
+	. = ..()
+	if(!isitem(parent))
```

Comment:
It makes sense to only limit this to items as any item can have a cell, but in that case can we call it the `crank_recharge` component or something?

Reply:
honestly i just kept it called gun_crank because i didnt test it on anything other than guns.